src/solver.rs

```diff
@@ -0,0 +1,275 @@
+use crate::c_bindings;
+use crate::coords::{LLHRadians, ECEF};
+use crate::navmeas::NavigationMeasurement;
+use crate::signal::GnssSignal;
+use crate::time::GpsTime;
+use std::borrow::Cow;
+use std::ffi;
+use std::fmt;
+
+#[derive(Clone, Debug)]
+pub struct GnssSolution(c_bindings::gnss_solution);
+
+impl GnssSolution {
+    pub fn pos_valid(&self) -> bool {
+        self.0.valid == 1
+    }
+
+    pub fn vel_valid(&self) -> bool {
+        self.0.velocity_valid == 1
+    }
```

Comment:
The C structures have these validity flags. Initially I was going to have the getter functions return optional values based on the validity flags, but that seemed a little janky for the user. As it stands if some data isn't valid it's value should be zero.

Reply:
Would it make sense to return an Option from the accessors instead of exposing these validity flags?

Reply:
Probably would be more "rust-y". I'll change this